code/WorkInProgress/Ported/policetape.dm

```diff
@@ -134,8 +138,8 @@
 			return 1
 
 		to_chat(user, "<span class='notice'>You start placing [src].</span>")
-		if(!do_mob(user, target, 3 SECONDS))
-			return 1
+		if(!do_after(user, target, 3 SECONDS))
+			return 0
```

Comment:
Ah. I should have noticed this problem while we were talking about it, but I didn't. This will make it so that you'll attack the door if you move away. Also, the other failure cases that return `1` will cause it to consume a charge.
I'd advise the same approach as for `attack_self()`: Move most of the logic here into a separate proc and call that one from here.

Reply:
My brain is an melt. I may need a walk through for this one....